apigentools/commands/generate.py

```diff
@@ -250,3 +261,20 @@ def run(self):
             self.write_dot_apigentools_info(language)
 
         return 0
+
+    def pull_repository(self, language):
+        output_dir = self.get_generated_lang_dir(language.language)
+        if self.args.use_https:
+            repo = REPO_HTTPS_URL.format(language.github_org, language.github_repo)
+        else:
+            repo = REPO_SSH_URL.format(language.github_org, language.github_repo)
+
+        try:
+            # Git doesn't allow you to clone into a non empty dir
+            # so create a temp dir and we'll move the content after
```

Comment:
So IMO if the repo is already there, we should:

* Make sure to switch branch to master
* Do `git pull`

Reply:
Any benefit over always doing a shallow clone and copying the contents?

Reply:
Also the directory being there doesn't gaurantee its the expected target git repo. This seemed safer to me.

Reply:
* copying over would overwrite any local changes that someone might have done locally; in the same way, pulling when there are conflicts would fail if any local changes conflict with what's being pulled
* it's true that the directory being there isn't necessarily the expected repo, but that's IMO something that the user should inspect instead of the tool just overwriting whatever is in it

All in all, I think we need to take extra caution to not overwrite any local changes that the user might have done.

Reply:
I agree about being cautious. We took this conversation offline and decided to:

* Not clone the repo by default
* Print an error message if the clone flag was set, but the output_dir isn't empty

We chose this primarily because we wanted to still allow users to make modifications and iterate on their spec locally without wiping out ignored files (like test files). But we want to ensure the entire process (if you do use this clone flag) can still push clean, unmodified commits to still provide reproducibility.